geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/collections/SizeableObject2ObjectOpenCustomHashMapWithCursor.java

```diff
@@ -29,52 +33,58 @@
  * The scan method provides the same guarantees as Redis's HSCAN, and in fact
  * uses the same algorithm.
  */
-public class Object2ObjectOpenCustomHashMapWithCursor<K, V>
-    extends Object2ObjectOpenCustomHashMap<K, V> {
+public class SizeableObject2ObjectOpenCustomHashMapWithCursor<K, V>
```

Comment:
since our redis data classes always use the class with K=byte[] and V=byte[] it seems like you could make this class be SizeableByteArray2ByteArrayOpenCustomHashMapWithCursor extends Object2ObjectOpenCustomHashMap<byte[], byte[]>
You could also do this for the HashSet class.
Then in the size code instead of using the ReflectionSingleObjectSizer you can just compute the size of a byte[] which is very simple. ReflectionSingleObjectSizer has to do extra work to handle any type of array. It seems like this would give you fast sizing.

Reply:
We currently have [a PR up](https://github.com/apache/geode/pull/6548) that uses a non-byte[] value in the CustomHashMap implementation, so I don't want to conflict with that by changing the underlying data structure.